### PR TITLE
Fix pacman stdout parsing in the Ansible module

### LINKED ISSUE #65237: `pacman` module fails when the locale is set
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
module `pacman`

##### ANSIBLE VERSION

```paste below
ansible 2.9.1
  config file = /home/lenaic/doc/devel/perso/lenaic_config/ansible.cfg
  configured module search path = ['/home/lenaic/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.8/site-packages/ansible
  executable location = /bin/ansible
  python version = 3.8.0 (default, Oct 23 2019, 18:51:26) [GCC 9.2.0]
```

##### CONFIGURATION

```paste below
ANSIBLE_PIPELINING(/home/lenaic/doc/devel/perso/lenaic_config/ansible.cfg) = True
CACHE_PLUGIN(/home/lenaic/doc/devel/perso/lenaic_config/ansible.cfg) = jsonfile
CACHE_PLUGIN_CONNECTION(/home/lenaic/doc/devel/perso/lenaic_config/ansible.cfg) = ~/.ansible/tmp/facts_cache
CACHE_PLUGIN_TIMEOUT(/home/lenaic/doc/devel/perso/lenaic_config/ansible.cfg) = 86400
DEFAULT_BECOME(/home/lenaic/doc/devel/perso/lenaic_config/ansible.cfg) = True
DEFAULT_BECOME_ASK_PASS(/home/lenaic/doc/devel/perso/lenaic_config/ansible.cfg) = True
DEFAULT_BECOME_METHOD(/home/lenaic/doc/devel/perso/lenaic_config/ansible.cfg) = su
DEFAULT_FORCE_HANDLERS(/home/lenaic/doc/devel/perso/lenaic_config/ansible.cfg) = True
DEFAULT_GATHERING(/home/lenaic/doc/devel/perso/lenaic_config/ansible.cfg) = smart
DEFAULT_HOST_LIST(/home/lenaic/doc/devel/perso/lenaic_config/ansible.cfg) = ['/home/lenaic/doc/devel/perso/lenaic_config/inventory.ini']
DEFAULT_REMOTE_USER(/home/lenaic/doc/devel/perso/lenaic_config/ansible.cfg) = lenaic
DEFAULT_VAULT_PASSWORD_FILE(/home/lenaic/doc/devel/perso/lenaic_config/ansible.cfg) = /home/lenaic/doc/devel/perso/lenaic_config/.vault_password
RETRY_FILES_ENABLED(/home/lenaic/doc/devel/perso/lenaic_config/ansible.cfg) = False
```

##### OS / ENVIRONMENT

Arch Linux

##### STEPS TO REPRODUCE

```shell
LANG=fr_FR.utf8 ansible localhost -C -D -m pacman -a "name=ansible"
```

##### EXPECTED RESULTS

```
localhost | SUCCESS => {
    "changed": false,
    "msg": "package(s) already present"
}
```

##### ACTUAL RESULTS

```paste below
localhost | FAILED! => {
    "changed": false,
    "msg": "get_name: fail to retrieve package name from pacman output"
}
```

### PR DESCRIPTION
**Pull request by L3n41c**
##### SUMMARY

`pacman` output is localized and the Ansible module is parsing its output.
So, we need to force the locale.

Fixes #65237

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

module `pacman`

### CODE REVIEW

File: lib/ansible/modules/packaging/os/pacman.py
Comment (tchernomax) [inline lib/ansible/modules/packaging/os/pacman.py]: I think the following should be sufficient:
```
module.run_command_environ_update = dict(LC_ALL='C')
```
because https://pubs.opengroup.org/onlinepubs/7908799/xbd/envvar.html :
> The value of the LC_ALL environment variable has precedence over any of the other environment variables starting with LC_ (LC_COLLATE, LC_CTYPE, LC_MESSAGES, LC_MONETARY, LC_NUMERIC, LC_TIME) and the LANG environment variable.

**Review by tchernomax — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/packaging/os/pacman.py
Comment (L3n41c) [reply lib/ansible/modules/packaging/os/pacman.py]: Indeed. Good catch. Fixed.
There are some other modules that could benefit from this simplification as well.

**Comment by L3n41c** [comment]
This PR currently target the `devel` branch but it fixes an issue which is already present with Ansible 2.9.
Should I target a different branch ? Or back port this change later ?

**Review by tchernomax — APPROVED** [review]
LGTM

> Should I target a different branch ? Or back port this change later ?

I don't know… I think it's up to the ansible core team to decide.

File: lib/ansible/modules/packaging/os/pacman.py
Comment (Akasurde) [inline lib/ansible/modules/packaging/os/pacman.py:436]: Could you please add a changelog entry for this change [here](https://github.com/ansible/ansible/tree/devel/changelogs/fragments) so that this change can be backported to 2.9 branch?

You can find the documentation on [How to backport merged PR](https://docs.ansible.com/ansible/latest/community/development_process.html?backporting-merged-prs#backporting-merged-prs)

Let me know once you are done with this change.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/packaging/os/pacman.py
Comment (L3n41c) [reply lib/ansible/modules/packaging/os/pacman.py:436]: @Akasurde : I added a changelog fragment.

I can submit the backport PR against 2.9 once this one is one is merged on devel.

**Review by Akasurde — APPROVED** [review]
LGTM

**Comment by Akasurde** [comment]
@L3n41c Thanks for the contribution. @tchernomax Thanks for the review.

File: lib/ansible/modules/packaging/os/pacman.py
Comment (L3n41c) [reply lib/ansible/modules/packaging/os/pacman.py:436]: Backport to 2.9 is in #66045.

**Pull request merged**
